test/acceptance/test/profiles_test.go

```diff
@@ -129,6 +129,29 @@ Namespace: %s`, clusterName, namespace)))
 			resp, statusCode, err = kubernetesDoRequest(namespace, clusterName+"-"+profileName, "9898", "/healthz", clientSet)
 			return statusCode
 		}, "120s", "1s").Should(Equal(http.StatusOK))
+
+		By("Updating the version of the installed profile")
+		time.Sleep(3 * time.Minute)
+		Eventually(func() string {
+			stdOut, stdErr = runCommandAndReturnStringOutput(fmt.Sprintf("%s update profile --name %s --version 6.0.0 --namespace %s --cluster %s --config-repo %s --auto-merge", gitopsBinaryPath, profileName, namespace, clusterName, appRepoRemoteURL))
+			return stdErr
+		}, "120s", "5s").Should(BeEmpty())
```

Comment:
Can we just remove the sleep and have the `Eventually` run over a long period of time instead? Less sleeps the better :smile:

Reply:
I reduced the `sleep` to 3 minutes and having the `Eventually` be `120s` works. I think using `sleep` is justified since we know that this command will definitely not be successful in the first 3-5 minutes after doing `add profile`. Polling in those first few minutes will only count towards the rate limit since we're not doing [conditional requests](https://docs.github.com/en/rest#conditional-requests)

Reply:
scratch that - I just moved it all to the `Eventually`s!